tpchgen/src/generators.rs

```diff
@@ -266,17 +281,57 @@ impl<'a> Iterator for RegionGeneratorIterator<'a> {
     }
 }
 
+/// A Part Manufacturer, formatted as "Manufacturer#<n>"
```

Comment:
this is an example of what I think @scsmithr  was suggesting in https://github.com/clflushopt/tpchgen-rs/pull/37#issuecomment-2736851885

Now the display logic is in a new type wrapper so that the consumer of a `PartManufacturer` can decide when/where to format the field